### ISSUE TITLE
copy/paste not working if content is not english text or filename

### ISSUE THREAD

**Issue by rzateev** [neutrinolabs/xrdp#339]
I use xubuntu 14.04 as server.
Cleint is Windows 8 or CentOS 6.7 (freerdp).

I  builde XRDP from the master source with XORG drivers.

Copy/paste text work perfectly when connent is english text or numerals. Also work copy/paste files if it name content english symbol.

I copy from client to server.

But if I try copy/paste text with content of RUSSIAN symbols it does not work - paste is empty.
![errorcopyfilewithrussiansymbolslist](https://cloud.githubusercontent.com/assets/17544153/13485033/e9c94a2c-e135-11e5-8c07-b13d53c0ffe1.png)

![errorcopyfilewithrussiansymbols](https://cloud.githubusercontent.com/assets/17544153/13484937/0b298f48-e135-11e5-8877-7b9880c0d164.png)

What can i do to fix IT?

**Comment by metalefty** [comment]
It works for me even with Japanese symbols. Using f422461.
https://youtu.be/DyXTMlU6b0g

Try this and let me know the result.
1. Copy a file with Russian symbols to clipboard in client side
2. Paste it in server side (will fail)
3. `ls ~/thinclient_drives/.clipboard`
4. `cp ~/thinclient_drives/.clipboard/your_name_with_with_russian_symbols.jpg /tmp`

**Comment by rzateev** [comment]
I used brach with previous commit  https://github.com/neutrinolabs/xrdp/commit/5af0d06d1e184a06c25bc044aa9a82921c2f59b5.


A little later, I try the latest build and say you result

**Comment by rzateev** [comment]
I try the last version of branch master - past russian symbols still does not work.
After copy text with russian symbols command
`ls ~/thinclient_drives/.clipboard`
result is empty.

Paste english text is work.

After copy file "2- WithRussianSymbolsЯЯЯЯ.jpg" to buffer on client:
`ls ~/thinclient_drives/.clipboard` 
result is empty
Paste with name "2- WithRussianSymbolsЯЯЯЯ.jpg" as result has name "2- WithRussianSymbols.jpg"

Command
`cp "~/thinclient_drives/.clipboard/2- WithRussianSymbolsЯЯЯЯ.jpg" /tmp` 
as result: file does not exist.

If I understand correctly , in my case not worked clipboard not through the file `~/thinclient_drives/.clipboard` (but it present)

This is commands what i used to install xrdp with xorg drivers (i use session "Session manager"):

```
#/bin/bash

# Get packages
sudo apt-get install libx11-dev libxfixes-dev libssl-dev libpam0g-dev libtool libjpeg-dev flex bison gettext autoconf libxml-parser-perl libfuse-dev xsltproc libxrandr-dev  python-libxml2 nasm xserver-xorg-dev fuse

#clone repo
git clone https://github.com/neutrinolabs/xrdp.git
cd xrdp
#git init
git submodule init
git submodule update

#lib  RFXcodec
cd librfxcodec
make
cd ..

#rdp module for xorg
cd xorgxrdp/
./bootstrap 
./configure 
make
sudo make install
cd ..

#xrdp itself
./bootstrap
./configure --enable-fuse  --enable-jpeg
make
sudo make install

#generate keys:
sudo xrdp-keygen  xrdp auto 2048
service xrdp start
```

**Comment by rzateev** [comment]
Perhaps it has something effect, but i don see the folder 
`/usr/X11R6/lib/X11/fonts/`
on my server that is present as option "FontPath" in file "/etc/X11/xrdp/xorg.conf"

**Comment by rzateev** [comment]
Hello, metalefty!
I try CentOS 6.7 with X11RDP-TH-Matic as you described on Youtube video.
It is really copy/paste russian symbols work.
But it use X11RDP backend.
I described that i used xorg-xrdp drivers.

With X11RDP backend i cannot resolve problem with keyboard layout. So I started to try alternatives
Please read issue https://github.com/neutrinolabs/xrdp/issues/337.
Maybe you how can i resolve my problem.

**Comment by metalefty** [comment]
@rzateev, I will read your reply later but I can answer this just now.

> On you video a see that you use non english layout. Are you use "us" keyboard layout? Can you try to reconnect to session an check that you can change layout from "jp" to "us"? 

I usually use the "us" layout but Japanese keyboard layout is used in the video.  The session in the video is xorg driver.  I have no keyboard layout problems even with reconnected session.  It perfectly works for me.

Anyway, let's talk about keyboard layout in #337.

**Comment by metalefty** [comment]
RH-Matic can build xorgxrdp drivers. I also tried xorgxrdp session on CentOS 7. It is working for me even with Russian symbols. Of course, no problem in x11rdp session.

This is what I did.
* `./X11RDP-RH-Matic.sh --with-xorg-driver --nox11rdp`
* `service xrdp start` or `systemctl start xrdp`
* Edit /etc/pam.d/xserver like this https://www.centos.org/forums/viewtopic.php?t=21185
* other session configurations for example startwm.sh

![cliprdr-russian](https://cloud.githubusercontent.com/assets/941609/13588853/83901b48-e516-11e5-9b9d-a962278ba76f.png)

I'll try Xubuntu/Ubuntu when I have time.

**Comment by rzateev** [comment]
Thanks metalefty. I will wait your test in Xibuntu/Ubuntu.

If it somehow help: 
I noticed that when I connected client drive , I do not see it on the files that have Russian names

![20160308_xubuntuclientdisks](https://cloud.githubusercontent.com/assets/17544153/13591552/3e159e3e-e51c-11e5-880c-afbac09005cd.png)

![20160308_windows8clientdisks](https://cloud.githubusercontent.com/assets/17544153/13591557/467979c4-e51c-11e5-8c74-52c39783f046.png)


I do not know how to check . But I assume that the problem is in the FUSE module - Xubuntu he does not understand Russian characters .

**Comment by metalefty** [comment]
Until I test it in Ubuntu, can you test if Japanese characters work in your environment? Both for clipboard file and drive redirection. 

Please copy/paste this for file or directory name.
* ロシアの冬は寒い
* 本日は晴天なり

**Comment by rzateev** [comment]
No, it does not work. I try to pase file with english file name - good. I try paste file in japanise file name - it does not work. After this my client drive was disconnect and clipboard 
 does not work even for English text

**Comment by rexio8** [comment]
I have the same problem with the current master branch on Xubuntu 16.04, but with polish letters like ę, ą, ś, ć, ń, ł, ó. English content in the clipbord gets transferred fine. The given Japanese characters also are not copied.

**Comment by metalefty** [comment]
multibyte text copy via clipboard is not working.
file transfer with multibyte filename via clipboard works for me.

**Comment by DoraemonYu** [comment]
o(∩_∩)o Hi. 
I have the same problem with the current master branch on Ubuntu 14.04. It does not work when copy files with Chinese Char.

**Comment by metalefty** [comment]
@rzateev @rexio8 @DoraemonYu 
Hello everyone, I think I identified 5575197 is causing this issue. I'm working on give you the fix.  Please be patient.  Quick workaround is to revert 5575197.  Can you try?

**Comment by metalefty** [comment]
I think it is fixed in devel branch by #410.  Now it is working for me.
